manifests/install.pp

```diff
@@ -16,7 +16,9 @@
   case $consul::install_method {
     'url': {
       if $::operatingsystem != 'darwin' {
-        ensure_packages(['unzip'], { 'before' => Staging::File['consul.zip'] })
+        if ! defined ( Package['unzip'] ) {
```

Comment:
can you also please remove these additional white spaces? Something like this is preferable: `if !defined(Package['unzip']) {`